### PR TITLE
fix: catch tokenProvider rejection

### PR DESCRIPTION
**Pull request by pyrho**
## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] Code changes are tested

## Description of the changes, What, Why and How?

When providing a `tokenProvider` function to `connectUser`, if the `tokenProvider` promise is rejected this will result in an uncaught promise rejection.

This fix adds a `try/catch` around the `await`ing of the user provided `tokenProvider` function as to gracefully handle the error.

## Changelog
- fix: catch tokenProvider rejection

### CODE REVIEW

File: test/unit/connection.js [hunk 1]
```diff
@@ -44,6 +44,18 @@ describe('connection', function () {
 
 	after(() => wss.close());
 
+	describe('Connection tokenProvider', () => {
+		it('should handle token provider rejection ', async () => {
+			const client = new StreamChat('apiKey', {
+				allowServerSideConnect: true,
+				baseURL: 'http://localhost:9999', // invalid base url
```
Comment (arnautov-anton) [inline test/unit/connection.js:51]:
```suggestion
```
Not sure why this is needed, could you elaborate?

**Review by arnautov-anton — COMMENTED** [review]
Thank you for your contribution, @pyrho! Looks like a no-brainer to me, what do you think, @MartinCupela and @petyosi?

File: test/unit/connection.js (hunk 1 above)
Comment (pyrho) [reply test/unit/connection.js:51]: I was not sure what the required parameters were for a proper `StreamChat` instantiation. If it works without it, we can remove it; iirc I just copy/pasted this from another test in this file :sweat_smile:

**Review by arnautov-anton — APPROVED** [review]
LGTM :taco:

**Pull request merged**